code-studio/src/js/components/progress/course_progress.jsx

```diff
@@ -22,18 +32,18 @@ var CourseProgress = React.createClass({
       if (stages[i].flex_category) {
         var flexRows = [], previous = stages[i].flex_category;
         for ( ; i < stages.length && stages[i].flex_category === previous; i++) {
-          flexRows.push(<CourseProgressRow stage={stages[i]} key={stages[i].name}/>);
+          flexRows.push(this.getRow(stages[i]));
         }
         rows.push(
-          <div className="flex-wrapper">
-            <div key={previous} className="flex-category">
+          <div className="flex-wrapper" key={i}>
+            <div className="flex-category">
               <h4>{previous}</h4>
               {flexRows}
             </div>
           </div>
         );
       } else {
-        rows.push(<CourseProgressRow stage={stages[i]} key={stages[i].name}/>);
+        rows.push(this.getRow(stages[i]));
         i++;
```

Comment:
I prefer `map` as well but this is a weird split `for` loop that can add rows in both the inner and outer loop.  Don't worry, this weirdness will go away 😄 — the final mocks won't have flex groups next to non-flex stages.